src/snowflake/snowpark/functions.py

```diff
@@ -377,9 +377,16 @@ def lit(
             build_builtin_fn_apply(ast, "lit", literal, datatype)
 
     if isinstance(literal, Column):
-        return literal
-
-    return Column(Literal(literal, datatype=datatype), _ast=ast, _emit_ast=_emit_ast)
+        # Create new Column, and assign expression of current Column object.
+        # This will encode AST correctly.
+        c = Column("", _emit_ast=False)
```

Comment:
just to be safe, can we enable this code path only when `_emit_ast=True`

Reply:
which concern do you have? `Column("", _emit_ast=False)` should be side-effect free code, right?

Reply:
the only concern is to stay on a old code path. All ast related code changes should follow original codepath when `_emit_ast=False`